pkg/api/steve/supportconfigs/generator.go

```diff
@@ -24,34 +25,114 @@ import (
+				Namespace: cspNamespace,
+			},
+			User:   userInfo.GetName(),
+			Groups: userInfo.GetGroups(),
+			Extra:  convertExtra(userInfo.GetExtra()),
+			UID:    userInfo.GetUID(),
+		},
+	}, metav1.CreateOptions{})
+	if err != nil {
+		return false, fmt.Errorf("failed to create sar %s", err)
+	}
+	if !response.Status.Allowed {
+		return false, nil
+	}
+	return true, nil
+}
+
+// getCSPConfig gets the configmap produced by the csp-adapter returns an error if not able to produce the map. Will return
+// an errNotFound if the map is not found at all
+func (c *CSPConfig) getCSPConfig(cspNamespace string) (map[string]interface{}, error) {
+	cspConfigMap, err := c.ConfigMaps.GetNamespaced(cspNamespace, cspAdapterConfigmap, metav1.GetOptions{})
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			return nil, errNotFound
+		}
+		return nil, err
+	}
+	retVal := map[string]interface{}{}
+	err = json.Unmarshal([]byte(cspConfigMap.Data["data"]), &retVal)
+	if err != nil {
+		return nil, err
+	}
+
+	return retVal, nil
+}
+
+// Handler implements http.Handler - and serves supportconfigs (tar file which contains support relevant information)
+type Handler struct {
+	adapterUtil cspAdapterInterface
+	cspConfig   cspConfigInterface
 }
 
 // NewHandler creates a handler using the clients defined in scaledContext
 func NewHandler(scaledContext *config.ScaledContext) Handler {
-	return Handler{
-		ConfigMaps:           scaledContext.Core.ConfigMaps(cspadapter.ChartNamespace),
+	cspConfig := &CSPConfig{
+		ConfigMaps:           scaledContext.Core.ConfigMaps(metav1.NamespaceAll),
 		SubjectAccessReviews: scaledContext.K8sClient.AuthorizationV1().SubjectAccessReviews(),
-		adapterUtil:          cspadapter.NewChartUtil(scaledContext.Wrangler.RESTClientGetter),
+	}
+	return Handler{
+		adapterUtil: cspadapter.NewChartUtil(scaledContext.Wrangler.RESTClientGetter),
+		cspConfig:   cspConfig,
 	}
 }
 
 // ServerHTTP implements http.Handler - attempts to authenticate/authorize the user. Returns a tar of support information
 // if the user can get the backing configmap in the adapter namespace
 func (h *Handler) ServeHTTP(writer http.ResponseWriter, request *http.Request) {
-	authorized, err := h.authorize(request)
+	var cspChartNamespace string
+	var cspChartName string
+
+	if usePAYG, ok := request.URL.Query()["usePAYG"]; ok && usePAYG[0] == "true" {
+		// if usePAYG query parameter exist that means we are using the new Pay-As-You-Go offering CSP billing adapter
+		cspChartNamespace = cspadapter.PAYGChartNamespace
+		cspChartName = cspadapter.PAYGChartName
+	} else {
+		// if userPAYG query parameter does exist that means we are using the old Managed License offering CSP adapter
+		cspChartNamespace = cspadapter.MLOChartNamespace
+		cspChartName = cspadapter.MLOChartName
+	}
+	authorized, err := h.cspConfig.authorize(cspChartNamespace, request)
 	if err != nil {
 		util.ReturnHTTPError(writer, request, http.StatusForbidden, http.StatusText(http.StatusForbidden))
```

Comment:
You respond with the same status code and body when there is an error determining if user is authorized and when there isn't, and the user is simply not authorized. Isn't the first a case of an InternalServerError?

Reply:
This pertains to existing code, my mistake. Resolving...